static/js/message_edit_history.js

```diff
@@ -72,18 +72,28 @@ export function fetch_and_render_message_history(message) {
             ui_report.error(
                 $t_html({defaultMessage: "Error fetching message edit history"}),
                 xhr,
-                $("#message-history-error"),
+                $("#dialog_error"),
             );
         },
     });
 }
 
 export function show_history(message) {
     const rendered_message_history = render_message_history_modal();
-    $("#message_feed_container").append(rendered_message_history);
 
-    fetch_and_render_message_history(message);
-    overlays.open_modal("#message-edit-history", {autoremove: true});
+    dialog_widget.launch({
+        html_heading: $t_html({defaultMessage: "Message edit history"}),
+        html_body: rendered_message_history,
+        html_submit_button: $t_html({defaultMessage: "Close"}),
+        id: "message-edit-history",
+        on_click: () => {},
+        close_on_submit: true,
+        focus_submit_on_open: true,
+        single_footer_button: true,
+        post_render: () => {
+            fetch_and_render_message_history(message);
+        },
+    });
```

Comment:
Instead of passing "Close" for `html_submit_button`, I think we should show Close/Cancel button unconditionally since all modals will have it and then hide the submit button by according to `hide_submit_button` variable. I think this would be more cleaner and readable.

Reply:
I think that makes sense; let's do that as a separate follow-up project though; this has enough moving parts already :)